internal/services/phone/site/site_resource.go

```diff
@@ -0,0 +1,544 @@
+package site
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/folio-sec/terraform-provider-zoom/internal/provider/shared"
+	"github.com/hashicorp/terraform-plugin-framework-validators/int32validator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/objectdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/objectplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/samber/lo"
+)
+
+var (
+	_ resource.Resource                = &tfResource{}
+	_ resource.ResourceWithConfigure   = &tfResource{}
+	_ resource.ResourceWithImportState = &tfResource{}
+)
+
+func NewPhoneSiteResource() resource.Resource {
+	return &tfResource{}
+}
+
+type tfResource struct {
+	crud *crud
+}
+
+func (r *tfResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	data, ok := req.ProviderData.(*shared.ProviderData)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected ProviderData Source Configure Type",
+			fmt.Sprintf("Expected *provider.ProviderData, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+	r.crud = newCrud(data.PhoneClient)
+}
+
+func (r *tfResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_phone_site"
+}
+
+func (r *tfResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: `Manages a site within Zoom Phone.
+
+## API Permissions
+
+The following API permissions are required in order to use this resource.
+This resource requires the ` + strings.Join([]string{
+			"`phone:read:site:admin`",
+			"`phone:write:site:admin`",
+			"`phone:update:site:admin`",
+			"`phone:delete:site:admin`",
+			"`phone:read:list_sites:admin`",
+			"`phone:read:list_emergency_addresses:admin`",
```

Comment:
ops, this is not necessary

Reply:
https://github.com/folio-sec/terraform-provider-zoom/pull/112/commits/ccb585ae0ef02014bcf1c9914511b829e5774584